src/remediations/fifi.js

```diff
@@ -247,19 +247,22 @@ function defineRHCEnabledExecutor (satellites, smart_management, rhc_enabled) {
             satellites.push({id: null, systems: partitionedSystems[0], type: 'RHC', rhcStatus: (rhc_enabled) ? CONNECTED : DISABLED});
         }
 
-        satlessExecutor.systems = partitionedSystems[1];
+        satlessExecutor.systems = _.filter(partitionedSystems[1], system => _.isUndefined(system.rhc_client) && !system.marketplace);
+        if (_.isEmpty(satlessExecutor.systems)) {
+            // If there are no "satless" executors remove this executor from list
+            satellites = _.difference(satellites, satlessExecutor);
```

Comment:
Does this actually work? The docs for `_.difference` says it should be used on two arrays.